Nightwatch/Service.fs

```diff
@@ -0,0 +1,118 @@
+module Nightwatch.Service
+
+open System
+open System.Threading
+open System.Threading.Tasks
+
+open FSharp.Control.Tasks
+open Microsoft.Extensions.Hosting
+open Serilog
+
+open Microsoft.Extensions.DependencyInjection
+open Nightwatch
+open Nightwatch.Core
+open Nightwatch.Core.Environment
+open Nightwatch.Core.FileSystem
+open Nightwatch.Core.Network
+open Nightwatch.ResourceConfiguration
+open Nightwatch.Resources
+
+type ProgramInfo =
+    { version : Version }
+
+let private logFullVersion (logger : ILogger) programInfo =
+    logger.Information("Nightwatch v. {0}", programInfo.version)
+    logger.Information("Config file format v. {0}", ResourceConfiguration.configFormatVersion)
+
+let private resourceFactories = [| Http.factory Http.system; Shell.factory Process.system |]
+
+let private configureResourceRegistry (logger : ILogger) =
+    let names = resourceFactories |> Seq.map (fun f -> f.resourceType)
+    logger.Information("Available resources: {0}", String.Join(", ", names))
+    Registry.create resourceFactories
+
+let private readResources fs config factories : Async<Result<Resource [], InvalidConfiguration []>> =
+    let splitResults seq =
+        let chooseOk = function
+        | Ok x -> Some x
+        | Error _ -> None
+
+        let chooseError = function
+        | Ok _ -> None
+        | Error x -> Some x
+
+        let isOk = chooseOk >> Option.isSome
+
+        let (results, errors) =
+            seq
+            |> Seq.toArray
+            |> Array.partition isOk
+        Array.choose chooseOk results, Array.choose chooseError errors
+
+    async {
+        let! resources = Async.AwaitTask <| ResourceConfiguration.read factories fs config
+        let (results, errors) = splitResults resources
+        return
+            if Seq.isEmpty errors
+            then Ok results
+            else Error errors
+    }
+
+let private createScheduler resources =
+    let schedule = Scheduler.prepareSchedule resources
+    task {
+        let! scheduler = Scheduler.create()
+        do! Scheduler.configure scheduler schedule
+        return scheduler
+    }
+
+let startService logger programInfo env fs (configFilePath : Path) =

```

Comment:
Please, add type annotations.

Reply:
Hm that wasn't intended to be public. I'll `private` it :)